invenio_app_ils/records_relations/retriever.py

```diff
@@ -0,0 +1,239 @@
+
+        # get any extra metadata stored in the sibling for the relation with
+        # this record
+        sibling_metadata = RelationsExtraMetadata.get_extra_metadata_from(
+            sibling, relation_type, record_pid.pid_value, record_pid.pid_type
+        )
+        if (
+            sibling_metadata
+            and "pid_value" in sibling_metadata
+            and "pid_type" in sibling_metadata
+        ):
+            # the extra metadata in the sibling has pid_value and pid_type of
+            # this record. Replace it with the sibling ones.
+            sibling_metadata["pid_value"] = sibling_pid.pid_value
+            sibling_metadata["pid_type"] = sibling_pid.pid_type
+
+        metadata.update(sibling_metadata)
+        return metadata
+
+    def _build_relation_obj(self, sibling_pid, relation_type):
+        """Return the relation object with metadata."""
+        pid_value = sibling_pid.pid_value
+        pid_type = sibling_pid.pid_type
+
+        r = self.build_relations_object(pid_value, pid_type, relation_type)
+
+        # fetch the sibling to retrieve useful metadata (title, ...) and also
+        # any optional extra metadata for this relation
+        from invenio_app_ils.records.api import IlsRecord
+        sibling = IlsRecord.get_record_by_pid(pid_value, pid_type=pid_type)
+
+        metadata = self._get_extra_metadata(
+            self.record, sibling, relation_type
+        )
+        # copy the extra metadata into the relation, if any
+        r.update(metadata)
+
+        # copy relevant fields from sibling into the relation
+        relevant_fields = self.get_relevant_fields_from(sibling)
+        r.update(relevant_fields)
+        return r
+
+    def get(self):
+        """Get all sibling relations with the current record."""
+        relations = {}
+        for relation_type in current_app.config["SIBLINGS_RELATION_TYPES"]:
+            sr = SiblingsRelation(relation_type)
+            name = relation_type.name
+
+            for sibling_pid in sr.all(self.record.pid):
+                r = self._build_relation_obj(sibling_pid, name)
+                relations.setdefault(name, [])
+                relations[name].append(r)
+        return relations
+
+
+class SequenceRetriever(RelationObjectBuilderMixin):
+    """Retrieve relations of type Sequence."""
+
+    ORDER_FIELD_NAME = "relation_order"
+    ORDER_VALUE_NEXT = "continues"
+    ORDER_VALUE_PREVIOUS = "is_continued_by"
+
+    def __init__(self, record):
+        """Constructor."""
+        self.record = record
+
+    def _build_relation_obj(self, related_pid, relation_type):
+        """Return the relation object with metadata."""
+        pid_value = related_pid.pid_value
+        pid_type = related_pid.pid_type
+
+        r = self.build_relations_object(pid_value, pid_type, relation_type)
+
+        # fetch the sequence to retrieve useful metadata (title, ...)
+        from invenio_app_ils.records.api import IlsRecord
```

Comment:
This one, like my previous comment too.

Reply:
it will cause circular import from what I understood